### PR TITLE
Generate: slow assisted generation test

### PR DESCRIPTION
**Pull request by gante**
# What does this PR do?

`test_assisted_decoding_matches_greedy_search` fails once in a while, which blocks development. This PR removes the blocker by moving it to a slow test. 

Why a slow test (and not redesign the test or add the flaky decorator)?
1. It is impossible to remove at 100% the non-determinism in this test. Some form of masking has to be used by design, which means that there is always a chance for the generations to diverge. When the generated sequences do diverge, the scores should be very similar at the step they diverge, as they are caused by the very small values within the numerical attention masks.
2. I've tried to add the check above (when the sequences diverge, the scores should be similar)... but some models still failed that check quite hard when the sequences didn't match. Some well-established models can run it without observed failures (e.g. 10k runs on GPT2 = 0 sequence mismatches). Others, especially roberta-based models, fail at a high rate. This means I should explore this further.
3. Since there is something to be explored, I believe the slow decorator is more appropriate: we can track failures without risking red CI on PRs.

### CODE REVIEW

File: tests/models/roberta/test_modeling_roberta.py [hunk 1]
```diff
@@ -397,10 +397,6 @@ class RobertaModelTest(ModelTesterMixin, GenerationTesterMixin, PipelineTesterMi
     )
     fx_compatible = True
 
-    @unittest.skip(reason="Fix me @gante")
-    def test_assisted_greedy_search_matches_greedy_search(self):
```
Comment (gante) [inline tests/models/roberta/test_modeling_roberta.py:401]: (this one was skipped because it was flaky)

**Review by ydshieh — APPROVED** [review]
I know that we want to make PR CI clean, but I am not a super fan to see the daily CI report being flaky neither. It has been clear that it's hard to focus on the daily CI report, although things get a bit better with a recent PR to add diff between 2 PRs.

OK for me, and we can discuss/find a way to deal with this flaky situation 🔥 .

File: tests/generation/test_utils.py [hunk 2]
```diff
@@ -1457,6 +1457,7 @@ def test_contrastive_generate_dict_outputs_use_cache(self):
             for output in (output_contrastive, output_generate):
                 self._check_outputs(output, input_ids, model.config, use_cache=True)
 
+    @slow  # TODO(Joao): remove this. Some models (e.g. data2vec, xcom, roberta) have an error rate between 1 and 10%.
```
Comment (ydshieh) [inline tests/generation/test_utils.py:1460]: Just wondering, since this test uses batch size 1 and short sequence, could we run it 10 times and check similar to #22996 ?

**Review by sgugger — COMMENTED** [review]
Does the error stay with non-random models? Maybe the test will be less flaky if done on a pretrained checkpoint (and since we are passing it at slow it would be ok to use those).

**Review by sgugger — APPROVED** [review]
Good for me to pass them as slow in any case, while investigating this further.

**Comment by gante** [comment]
> Maybe the test will be less flaky if done on a pretrained checkpoint

Definitely! However, I think there is a deeper problem here, the logits diverge way more than I'd expect on some models, and it's odd that those models rely on the same base code (roberta). After I finish preparing the release for assisted generation, I'll get back to sorting related bugs
